### DISCUSSION TITLE
jsPsych.init: jsPsych is not defined error

### DISCUSSION THREAD

**Discussion by RachaelBryant** [jspsych/jsPsych#764] [Q&A] [answered]
Hi! I have been struggling over this for almost a full week and I just don't know what's wrong! (I am a MSc student who suddenly has to make an online experiment for her dissertation project because in-person is no longer allowed and is learning coding in a rush)

This is the only thing (presumably) stopping my code from working.
![jsPsych init](https://user-images.githubusercontent.com/65221687/84966996-cb0f9d00-b0c7-11ea-8a6d-cad95d79d3a1.PNG)

Anyone have any idea why this is happening to me? I've been searching frantically but haven't found any answers!

Thank you~

**Comment by vijaymarupudi** [comment]
Can you post the entire html of your page? It is likely that you did not load jsPsych

https://www.jspsych.org/tutorials/hello-world/#step-5-import-the-jspsych-library

**Comment by RachaelBryant** [comment]
I have. If only it was that easy a solution!




```html
<!DOCTYPE html>
<html>
<head>
    <title>Typing Twisters</title>
    <script src="js\jspsych.js"type="text/js"></script>
    <script src="js\plugins\jspsych-html-button-response.js" type="text/js"></script>
    <script src="js\plugins\jspsych-instructions.js" type="text/js"></script>
    <script src="js\plugins\jspsych-survey-multi-select.js" type="text/js"></script>
    <script src="js\examples\jspsych-survey-likert.html" type="text/js"></script>
    <link href="css\jspsych.css" rel="stylesheet" type="text/css">
</head>
<body><div id="jspsych-target"></div></body>

<script>
    //welcome
    var welcome_block = {
      type: "html-button-response",
      stimulus:"<p>Welcome to this experiment</p>",
      choices:['continue']

  };

    //Instructions
    var instruction_block1 = {
      data: {
        screen_id: "instructions"
      },
      type: "instructions",
      pages: [
        "<p>Please answer the following questions to the best of your ability</p>",
      ],
      show_clickable_nav: true
    };

    //survey

    var survey_page1 = {
      type: 'survey-text',
      questions: [
        {prompt: 'How old are you?', columns: 3, required: true, name: 'Age'},
        {prompt: 'What country are you from/a citizen of?', placeholder: 'Country', columns: 50, name: 'BirthLocation'}
      ],
      randomize_question_order: true
    };

    var multi_select_block = {
        type: 'survey-multi-select',
        questions: [
          {
            prompt: "Are you right-handed, left-handed or ambidexturous?",
            options: ["Right-handed", "Left-handed", "ambidexturous"],
            horizontal: true,
            required: true,
            name: 'hand'
          }
        ],
        randomize_question_order: true
    };

    //Practice
    var practice_trial1 = {
      stimulus: "five tide type fine",
      choices: "ALL_KEYS",
      trial_duration: null,
    };

    var practice_trial2 = {
      stimulus: "five tide tap mine",
      choices: "ALL_KEYS",
      trial_duration: null,
    };


    //Experiment instructions
    var instruction_block2 = {
      data: {
        screen_id: "instruction2"
      },
      type: "instructions",
      pages: [
        "<p>Please type the sequence of words that you see to the best of your ability.</p>"+
        "<p>The practice trials are not timed so please press any key to move on to the next page.</p>",
      ],
      show_clickable_nav: true
    };

    //Experimental trials

    var survey_trial1= {
      type: 'survey-text',
      questions: [
        {prompt: "cow bill bat cap"}

      ],
      randomize_question_order: true
    };

    var survey_trial2= {
      type: 'survey-text',
      questions: [
        {prompt: 'rib mess mad rag'}
      ]
    };


    //Typing Survey Questions
    var scale = ["Strongly Disagree", "Disagree", "Neutral", "Agree", "Strongly Agree"];

    var likert_trial = {
        type: 'survey-likert',
        questions: [
          {prompt: "I typically make many mistakes when typing", name: 'TypicalTyping', labels: scale, required: true},
          {prompt: "Today I made more typing mistakes than usual.", name: 'TodayTyping', labels: scale, required: true}
        ],
    };

    //Debrief
    var debrief = {
      type: "html-button-response",
      stimulus:"<h1>Thank you for participating!</h1>",
      choices: ["Click any key to exit the experiment."]

  };


    //timeline
    var timeline= []
    timeline.push(welcome_block),
    timeline.push(instruction_block1),
    timeline.push(survey_page1),
    timeline.push(multi_select_block),
    timeline.push(practice_trial1),
    timeline.push(practice_trial2),
    timeline.push(instruction_block2),
    timeline.push(survey_trial1),
    timeline.push(survey_trial2),
    timeline.push(likert_trial);


    jsPsych.init ({
      timeline: timeline,
      show_progress_bar: true,
      display_element: 'jspsych-target',
      on_finish: function() {
        jsPsych.data.displayData('csv')
      }
    });

  </script>
</html>
```html

**Reply by vijaymarupudi** [reply]
The solution is pretty easy I think. Notice how you are using backslashes for the urls. That's a windows thing, which the internet does not follow. Replace them with forward slashes.

**Reply by RachaelBryant** [reply]
I've tried that before, and just now I tried it again but I still have the same error message :/ I actually even checked it in Gorilla, which doesn't use the scripted plug-ins at the top and just relies on you uploading the plug-ins and the version jspsych you're using and it doesn't work there either. I just don't know what else to check :/ I feel like I must be missing a line of code somewhere to activate jsPsych.init somehow?

**Comment by vijaymarupudi** [comment]
Maybe there are more issues, but this is an issue for sure. Make sure you use forward slashes.

Can you give me an idea of your file layout of your experiment?

**Comment by vijaymarupudi** [comment]
Oh and additionally, move the script tag into the body tag after your target element.

**Comment by RachaelBryant** [comment, marked as answer]
This is my file layout. I also thought this might be an issue, which is why I checked with Gorilla, which doesn't rely on it. But the error was still there :/
When I moved script into the body, it turned off the end body and it won't seem to change back no matter what i do? 

![file layout2](https://user-images.githubusercontent.com/65221687/84968647-e381b680-b0cb-11ea-9de7-5aac102f412a.PNG)

**Reply by vijaymarupudi** [reply]
I see an error here. There is no `jspsych.js` in the `js/` folder. You need to use the `js/jsPsych-6.1.0/jspsych.js` path

**Reply by vijaymarupudi** [reply]
Also remove `text/js`, that's not necessary and might prevent this from working.

**Reply by RachaelBryant** [reply]
Thank you! This actually did something! There are quite a few more errors that came up after fixing this, but at least I don't have a completely blank page anymore! Thank you!

**Reply by vijaymarupudi** [reply]
Those errors are a separate issue, I would try debugging them and if you can't fix it, asking for help in a different thread!

**Comment by RachaelBryant** [comment]
Sorry haha, I just realized that the entire script part goes into the body. I did that feeling hopeful...and yet nothing again :/ same error. Just a blank page appears in my browser

**Comment by RachaelBryant** [comment]
I've even looked into jspsych.js file directly to look for JsPsych.init but it doesn't have anything directly on it other than settings? But again, I am a total beginner so I'm really not sure what my problem is. Everyone else seems to have no problem with jsPsych.init :/
